src/runtime_src/core/common/api/xrt_bo.cpp

```diff
@@ -729,83 +730,106 @@ namespace xrt {
 
 bo::
 bo(xclDeviceHandle dhdl, void* userptr, size_t sz, bo::flags flags, memory_group grp)
-  : handle(alloc(dhdl, userptr, sz, static_cast<xrtBufferFlags>(flags), grp))
+  : handle(xdp::native::profiling_wrapper(__func__, "xrt::bo",
+           alloc_userptr, dhdl, userptr, sz, static_cast<xrtBufferFlags>(flags), grp))
 {}
 
 bo::
 bo(xclDeviceHandle dhdl, size_t size, bo::flags flags, memory_group grp)
-  : handle(alloc(dhdl, size, static_cast<xrtBufferFlags>(flags), grp))
+  : handle(xdp::native::profiling_wrapper(__func__, "xrt::bo",
+           alloc, dhdl, size, static_cast<xrtBufferFlags>(flags), grp))
 {}
 
 bo::
 bo(xclDeviceHandle dhdl, xclBufferExportHandle ehdl)
-  : handle(alloc(dhdl, ehdl))
+  : handle(xdp::native::profiling_wrapper(__func__, "xrt::bo",
+	   alloc_import, dhdl, ehdl))
 {}
 
 bo::
 bo(const bo& parent, size_t size, size_t offset)
-  : handle(sub_buffer(parent.handle, size, offset))
+  : handle(xdp::native::profiling_wrapper(__func__, "xrt::bo",
+	   sub_buffer, parent.handle, size, offset))
 {}
 
 bo::
 bo(xrtBufferHandle xhdl)
-  : handle(get_boh(xhdl))
+  : handle(xdp::native::profiling_wrapper(__func__, "xrt::bo",
+	   get_boh, xhdl))
 {}
 
 size_t
 bo::
 size() const
 {
-  return handle->get_size();
+  return xdp::native::profiling_wrapper(__func__, "xrt::bo", [this]{
+    return handle->get_size();
+  }) ;
 }
 
 uint64_t
 bo::
 address() const
 {
-  return handle->get_address();
+  return xdp::native::profiling_wrapper(__func__, "xrt::bo", [this]{
+    return handle->get_address();
+  });
 }
 
 xclBufferExportHandle
 bo::
 export_buffer()
 {
-  return handle->export_buffer();
+  return xdp::native::profiling_wrapper(__func__, "xrt::bo", [this]{
+    return handle->export_buffer();
+  });
 }
 
 void
 bo::
 sync(xclBOSyncDirection dir, size_t size, size_t offset)
 {
-  handle->sync(dir, size, offset);
+  return xdp::native::profiling_wrapper(__func__, "xrt::bo",
+    [this, dir, size, offset]{
+      handle->sync(dir, size, offset);
+    });
 }
 
 void*
 bo::
 map()
 {
-  return handle->get_hbuf();
+  return xdp::native::profiling_wrapper(__func__, "xrt::bo", [this]{
+    return handle->get_hbuf();
+  });
 }
 
 void
 bo::
 write(const void* src, size_t size, size_t seek)
 {
-  handle->write(src, size, seek);
+  xdp::native::profiling_wrapper(__func__, "xrt::bo", [this, src, size, seek]{
+    handle->write(src, size, seek);
+  });
 }
 
 void
 bo::
 read(void* dst, size_t size, size_t skip)
 {
-  handle->read(dst, size, skip);
+  xdp::native::profiling_wrapper(__func__, "xrt::bo", [this, dst, size, skip]{
+    handle->read(dst, size, skip);
+  });
 }
 
 void
 bo::
 copy(const bo& src, size_t sz, size_t src_offset, size_t dst_offset)
 {
-  handle->copy(src.handle.get(), sz, src_offset, dst_offset);
+  xdp::native::profiling_wrapper(__func__, "xrt::bo",
+    [this, src, sz, src_offset, dst_offset]{
```

Comment:
Might be better to capture the reference for src to avoid shared_ptr copy.  `&src`

Reply:
Changed all wrapper calls to pass references in the lambda when references were passed in the original function